File: problem_4/python/problem4.py
#!/usr/bin/python

def is_palindrome(number):
	start = 0
	end = len(number) -1
	
	while start < end:		
		if number[start] != number[end]:
			return 0

		start +=1
		end -= 1

	return 1

max = 0
for i in xrange(100, 1000):
	for j in xrange(100, i):
		value = i * j
		if is_palindrome(str(value)) and value > max:
			max = value

print max

File: problem_4/js/problem4.js
function is_palindrome(str_number) {
	// JS strings are immutable.
	// To reverse it: 
	//   - transform it in an array
	//   - reverse the array 
	//   - join it again
	if (str_number === str_number.split('').reverse().join('')) {
		return true;
	}

	return false;
}

function largest_palindrome(start, end) {
	// Just to be sure start can be the maximum
	var max = start - 1;

	// Two cicles because we need all possible combinations
	for (var i = start; i < end; i++) {
		for (var j = start; j < i; j++) {
			value = i * j;
			if (is_palindrome(value.toString()) &&  value > max) {
				max = value
			}
		}
	}

	return max;
}

console.log(largest_palindrome(100, 1000));